%!TEX root = ../main.tex

\graphicspath{{../../figures/chapter_1/}}

\part{Pipeline}
\chapter{FISH-Quant} \label{chap:chapter_1}
\minitoc
\newpage


\section{A tool for biologists}

%~\cite{lecuyer_global_2007}
%~\cite{ALLALOU200958} BlobFinder simple pipeline (segment with h-maxima, detect spots with filter/threshold, assign spots to cell/nucleus, count)

%~\cite{shariff_automated_2010}
%~\cite{laux_interactive_2020}
%~\cite{ljosa_introduction_2009}

% reference dypfish

~\cite{savulescu_dypfish_2019}
We introduce a range of analytical techniques for quantitatively interrogating
single molecule RNA FISH data in combination with protein immunolabeling over time.
Strikingly, our results show that constraining cellular architecture reduces
variation in subcellular mRNA and protein distributions, allowing the
characterization of their localization and dynamics with high reproducibility

~\cite{savulescu_interrogating_2021}
Here, we present DypFISH, an approach to quantitatively investigate the
subcellular localization of RNA and protein. We introduce a range of analytical
techniques to interrogate single-molecule RNA fluorescence in situ hybridization
(smFISH) data in combination with protein immunolabeling. DypFISH is suited to
study patterns of clustering of molecules, the association of mRNA-protein
subcellular localization with microtubule organizing center orientation, and
interdependence of mRNA-protein spatial distributions

% reference bento

~\cite{mah_bento_2022}
Bentoâ€™s utility, we applied it to analyze spatial transcriptomics datasets generated by seqFISH+ (10k genes in ~200 fibroblast cells) and MERFISH
Bento ingests single-molecule resolution data from highly multiplexed
spatial transcriptomics imaging experiments, enabling visualization, exploration and analysis of subcellular
biology.

% reference battich, stoeger, etc...

~\cite{battich_image-based_2013}
We obtained 18 primary spot features that reflect the
relative localization of each spot in a single cell, with respect to both the cell and other spots

~\cite{stoeger_computer_2015}
(talking about ~\cite{battich_image-based_2013})
Therefore, we have previously developed and documented [4] an unsupervised
clustering scheme that uses selected cellular statistics to identify a small
number of main patterns in single cell subcellular transcript localization.
Briefly, this package uses the per-cell mean and standard deviation of the
single-transcript localization features to first identify a number of
different patterns, by clustering random subsets of cells, such that
the clusters are most reproducible. In a second step, it determines the
similarity of each single cell to each of the identified patterns.

To enable imagebased transcriptomics to reach its full potential, we developed
computer vision algorithms that build on and improve those currently used to
detect objects in confocal images. By using iterative watershedding we have
improved the segmentations of nuclei and cells. In addition, we describe how
to perform spot detection for transcript identification in an automated way for
thousands of images. Accurate detection of nuclear outlines, cell outlines, and
transcript molecules are essential for the correct quantification of a
high-dimensional multivariate feature space of each transcript and to reveal
bona fide novel properties of the spatial organization of the transcriptome [4].
The computer vision pipeline presented here complements our earlier work [4],
and can be used independently of transcripts in other image-based approaches.



\section{Big-FISH}


\subsection{Organization and principles}

\subsection{Tutorials}


\section{Sim-FISH}


\subsection{Spots simulations}

\subsection{Localization pattern simulation}


\section{ImJoy plugins}


\subsection{Detection API}

\subsection{Segmentation API}

\subsection{Analysis API}
