%!TEX root = ../main.tex

\graphicspath{{./figures/introduction/}}

\chapter{Image-based transcriptomics}
\label{ch:introduction}

\minitoc
\newpage

\section{Gene expression process}
\label{sec:gene_expression}

\subsection{Transcription}

\subsection{Translation}

\subsection{Transport mechanisms}

\subsection{\ac{mRNA} localization}

% paper racha (introduction)

how
- rna metabolism
- protein metabolism (mature protein or nascent protein)

why
- storage of untranslated rna
- local translation
- rapid cell division cycle (linked to local translation of cyclin B mRNAs)
- cell polarization
- cell motility (actin at the leading edge)
- axonal growth and synaptic plasticity of neurons
- assembly of protein complexes
- avoid proteins in wrong place

how
- targeting signal of nascent protein
- targeting from rna zip-code sequence (with RNA-binding proteins)
- direct transport on the cytoskeleton by molecular motors
- anchoring mechanism
- diffusion, trapping, degradation and local rna stabilization

claim
- lack a global view of local translation in the entire cellular space or at the genomic level
- need a systematic manner

examples
- pioneering study in Drosophila
- human cell lines

limitations
- information on RNA localization but did not directly investigate local translation as the encoded proteins were not detected

% Most mRNAs are distributed randomly throughout the cytoplasm, but some localize to specific
% subcellular areas (Blower, 2013; Bovaird et al., 2018; Eliscovich and Singer, 2017; Jung et al., 2014 for reviews).

% This phenomenon is linked to either RNA metabolism, when untranslated mRNAs are stored in P- bodies or other cellular structures (Hubstenberger et al., 2017),
% or to protein metabolism, when a protein is synthesized locally.

% Local translation has been observed from bacteria and yeast to humans (Blower, 2013; Jung et al., 2014; Eliscovich and Singer, 2017; Bovaird et al., 2018).

% It is commonly involved in the delivery of mature proteins to specific cellular compartments, while allowing local regulation, and this is involved in many processes.
% For instance, it contributes to patterning and cell fate determination during metazoan development, mainly through asymmetric cell division
% (Melton, 1987; Driever and Nu€sslein- Volhard, 1988).
% In Xenopus embryos, local translation of cyclin B mRNAs at the mitotic spindle is also believed to be important for the rapid cell
% division cycles occurring during early embryo- genesis (Groisman et al., 2000).
% In mammals, mRNA localization is involved in cell polarization and motility, mainly through the localization of actin and related mRNAs at the leading edge (Lawrence and Singer, 1986), and it is also involved in axonal
% growth and synaptic plasticity of neurons (Van Driesche and Martin, 2018).

% Importantly, local translation can also be linked to the
% metabolism of the nascent peptide rather than to directly localize the mature protein. For instance, translation of secreted proteins
% at the endoplasmic reticulum (ER) allows nascent pro- teins to translocate through the membrane to reach the ER lumen (Aviram and Schuldiner, 2017).
% Translation of mRNAs at specific sites may also be important for the assembly of protein complexes (Pichon et al., 2016) or
% to avoid the deleterious effects of releasing free proteins at inappropriate places (Mu€ller et al., 2013).

% RNA localization can be accomplished through several mech- anisms. In the case of secreted proteins, the nascent peptide serves as a
% targeting signal, via the signal recognition particle (SRP) and its receptor on the ER (Aviram and Schuldiner, 2017).

% In most other cases, targeting is an RNA-driven process (Blower, 2013; Jung et al., 2014; Eliscovich and Singer, 2017; Bovaird et al., 2018 for reviews).
% Localized mRNAs often contain a zip-code sequence, frequently located within their 30-UTR, which is necessary and sufficient to transport them to their destination.
% The zip code is recognized by one or several RNA-binding proteins (RBPs), and it drives the formation of a transport complex sometimes called locasome.
% This complex can be transported by centrosomes, endosomal vesicles, or other cellular structures (Blower, 2013; Jung et al., 2014; Eliscovich and Singer, 2017; Bovaird et al., 2018).

% However, direct transport on the cytoskeleton by molecular motors is a frequent mechanism (Blower, 2013; Bovaird et al., 2018; Eliscovich and Singer, 2017; Jung et al., 2014).

% Once at destination, an anchoring mechanism may limit diffusion away from the target site.

% Alternatives to these transport mechanisms include diffusion and trap- ping at specific locations and degradation coupled to local RNA stabilization.

% Localized mRNAs are often subjected to a spatial control of translation (Besse and Ephrussi, 2008).
% In the case of Ash1 mRNA in yeast and b-actin mRNA in neurons, translation is repressed during transport and
% is activated at their final location by phosphorylation-dependent mechanisms (Hu€ttelmaier et al., 2005; Paquin et al., 2007).
% This spatial regulation of translation provides an additional layer of control ensuring that mRNAs are translated only at the desired location.

% The first locally translated mRNAs were found by chance or using a candidate approach.
% Purification of cellular structures and localized RBPs have significantly increased the number of known localized mRNAs
% (Blower, 2013; Jung et al., 2014; Eliscovich and Singer, 2017; Bovaird et al., 2018).

% However, only specific compartments or RBPs were examined, and we currently lack a global view of local translation in the entire cellular space or at the genomic level.
% Few reports described attempts to characterize mRNA localization in a systematic manner.

% A pioneering study in Drosophila used whole-mount fluorescent in situ hybrid- ization (FISH) to analyze the localization of more than 2,000 mRNAs (Lecuyer et al., 2007).
% As many as 71\% of them had a non-random distribution, and a range of new localization patterns were observed.
% More recent reports confirmed that RNA localization is widespread during Drosophila development (Jam- bor et al., 2015; Wilk et al., 2016).

% However, it is not known whether this is also true in other organisms, particularly in humans. Few recent studies addressed this question in cell lines
% using the more sensitive single-molecule FISH technique (smFISH). Several thousands of mRNAs were analyzed, which
% showed a correlation of intracellular mRNA distribution with gene annotation (Battich et al., 2013; Chen et al., 2015; Eng et al., 2019; Xia et al., 2019).
% Specifically, these studies identified three groups of localized mRNAs, in the perinuclear area, the mitochondria, and the cell periphery,
% the latter being possibly linked to actin metabolism (Chen et al., 2015).

% These studies provided information on RNA localization but did not directly investigate local translation as the encoded proteins were not detected.
% Thus, we still lack a good understanding of the various functions played by local translation at the cellular level.

% In this study, we developed a smFISH screen to specifically address this issue. Using a set of 523 GFP-tagged cell lines spanning
% a variety of cellular functions and an approach that allows simultaneous visualization of mRNA and proteins,
% we found that local translation occurs at various unanticipated locations. In particular, we discovered specialized
% translation factories, where specific mRNAs are translated. These factories are remarkable in that they provide
% a unique mean to regulate the metabolism of nascent proteins and also create a fine granular compartmentalization of translation.

% https://www.sciencedirect.com/science/article/pii/S2589004221012670

\subsection{Protein localization}

\section{Imaging cells and \ac{RNA}s}
\label{sec:fish}

% paper racha (dual protein-rna localization screens)

% In order to simultaneously visualize mRNAs with their encoded proteins, we based our screen on a library of HeLa cell lines,
% each containing a bacterial artificial chromosome (BAC) stably integrated in their genome (Poser et al., 2008).
% Each BAC con- tains a GFP-tagged gene harboring all its regulatory sequences (promoter, enhancers, introns, and 50 and 30 UTRs; Figure 1A).
% The resulting mRNAs are, thus, identical to the endogenous mol- ecules in terms of sequence and isoform diversity, except for the added tag.
% Previous studies showed that such tagged genes are expressed at near endogenous levels and with the proper spatio- temporal pattern (Poser et al., 2008).
% Since the tagged mRNAs contain all the regulatory sequences, we hypothesized that they would localize like the endogenous ones,
% provided that the tag does not interfere with localization. Using BACs offers two advantages. First, a single smFISH probe set
% against the GFP sequence is sufficient to detect all the studied mRNAs. Sec- ond, using mild hybridization conditions,
% GFP fluorescence can be detected together with the smFISH signal (Fusco et al., 2003), and thus, both the mRNA and the encoded protein can be detected in the same cell.

\subsection{Fluorescent stains and \ac{GFP}}

\subsection{\ac{FISH}}

\subsection{\ac{smiFISH}}

\subsection{\ac{SeqFISH} and \ac{MERFISH}}

\section{Measuring images: from pixels to numbers}
\label{sec:computation_biology}

\subsection{MATLAB, java and python: a zoo of solutions}

\subsection{Detect, segment and analyze}

% unknown
~\cite{shariff_automated_2010}
~\cite{laux_interactive_2020}
~\cite{das_intracellular_2021}

% general pipeline
~\cite{mcquin_cellprofiler_2018}
~\cite{mueller_fish-quant_2013}
~\cite{de_chaumont_icy_2012}
~\cite{ershov_bringing_2021} % trackmate
~\cite{ljosa_introduction_2009}
~\cite{stoeger_computer_2015}
~\cite{perkel_starfish_2019}
~\cite{noauthor_mammalian_2020}
~\cite{eng_transcriptome-scale_2019}
~\cite{kamenova_co-translational_2019}
~\cite{liao_rna_2019}
~\cite{xia_spatial_2019}
~\cite{tsanov_smifish_2016}
~\cite{samacoits_computational_2018}
~\cite{battich_image-based_2013}
~\cite{savulescu_interrogating_2021}

~\cite{battich_image-based_2013}
We obtained 18 primary spot features that reflect the
relative localization of each spot in a single cell, with respect to both the cell and other spots
here we show that branched dnA technology combined with automated liquid handling, high-content imaging and quantitative image analysis allows highly reproducible quantification of transcript abundance in thousands of single cells at single-molecule resolution.
in addition, it allows extraction of a multivariate feature set quantifying subcellular patterning and spatial properties of transcripts and their cell-to-cell variability. this has multiple implications for the functional interpretation of cell-to-
cell variability in gene expression and enables the unbiased identification of functionally relevant in situ signatures
of the transcriptome without the need for perturbations. Because this method can be incorporated in a wide variety of high-throughput image-based approaches, we expect it to be broadly applicable.

~\cite{stoeger_computer_2015}
(talking about ~\cite{battich_image-based_2013})
Therefore, we have previously developed and documented [4] an unsupervised
clustering scheme that uses selected cellular statistics to identify a small
number of main patterns in single cell subcellular transcript localization.
Briefly, this package uses the per-cell mean and standard deviation of the
single-transcript localization features to first identify a number of
different patterns, by clustering random subsets of cells, such that
the clusters are most reproducible. In a second step, it determines the
similarity of each single cell to each of the identified patterns.

To enable imagebased transcriptomics to reach its full potential, we developed
computer vision algorithms that build on and improve those currently used to
detect objects in confocal images. By using iterative watershedding we have
improved the segmentations of nuclei and cells. In addition, we describe how
to perform spot detection for transcript identification in an automated way for
thousands of images. Accurate detection of nuclear outlines, cell outlines, and
transcript molecules are essential for the correct quantification of a
high-dimensional multivariate feature space of each transcript and to reveal
bona fide novel properties of the spatial organization of the transcriptome [4].
The computer vision pipeline presented here complements our earlier work [4],
and can be used independently of transcripts in other image-based approaches.

~\cite{battich_control_2015}
Here, we applied image-based transcriptomics, a highthroughput automated
single-molecule fluorescence in situ hybridization (sm-FISH) method that we
recently developed (Battich et al., 2013), which meets these requirements.
Using large-scale single-cell datasets acquired with this approach, we show
that cell-to-cell variability in cytoplasmic transcript abundance in human
adherent cells can be accurately predicted at the single-cell level with a
multivariate set of features that quantify properties of the cellular state
and microenvironment, and we experimentally verify some of the underlying
causality. We find that for most genes, the unexplained variability in cytoplasmic
transcript abundance approaches a limit of minimal stochasticity imposed by a
Poisson process. The few genes that deviate from this limit also show a high
amount of explained variability, suggesting high-level regulation rather than
high stochasticity. Through computational multiplexing, we uncover the existence
of multilevel transcript homeostasis in single cells to achieve specific
adaptation of transcript abundance to the cellular state and microenvironment,
according to function of the proteins they encode. Finally, we show that the
mammalian nucleus acts as a potent and global buffer to stochastic fluctuations
arising from bursts in gene transcription by temporally retaining transcripts
in the nucleus. This explains how cytoplasmic transcript abundance in mammalian
cells can be minimally stochastic, while deterministic variation is maintained.

% ML for detection ?
~\cite{khater_caveolae_2019}

\subsection{StarFISH: a common \ac{FISH} platform}

\subsection{FISH-Quant MATLAB}

\section{Contributions}
\label{sec:contributions}

\subsection{FISH-Quant v2}

\subsection{Quantifications of localization patterns}
